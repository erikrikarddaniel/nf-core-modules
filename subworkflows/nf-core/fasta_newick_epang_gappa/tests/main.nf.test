nextflow_workflow {

    name "Test Workflow FASTA_NEWICK_EPANG_GAPPA"
    script "../main.nf"
    config "./nextflow.config"

    workflow "FASTA_NEWICK_EPANG_GAPPA"

    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows/fasta_newick_epang_gappa"
    tag "fasta_newick_epang_gappa"
    tag "epang"
    tag "epang/place"
    tag "epang/split"
    tag "gappa"
    tag "gappa/examineassign"
    tag "gappa/examinegraft"
    tag "gappa/examineheattree"
    tag "hmmer"
    tag "hmmer/hmmbuild"
    tag "hmmer/hmmalign"
    tag "hmmer/eslalimask"
    tag "hmmer/eslreformat"
    tag "mafft"


    test("test_fasta_newick_epang_gappa_hmmer") {
        
        when {
            params {
                outdir = "$outputDir"
            }
            workflow {
                """
                input[0] = Channel.of(
                    [
                        meta: [ id: "hmmer" ],
                        data: [
                            alignmethod:  'hmmer',
                            queryseqfile: file("https://github.com/nf-core/test-datasets/raw/phyloplace/testdata/PF14720_3_sequences.faa", checkIfExists: true),
                            refseqfile:   file("https://github.com/nf-core/test-datasets/raw/phyloplace/testdata/PF14720_seed.alnfaa", checkIfExists: true),
                            refphylogeny: file("https://github.com/nf-core/test-datasets/raw/phyloplace/testdata/PF14720_seed.ft.LGCAT.newick", checkIfExists: true),
                            model:        "LG",
                            taxonomy:     file("https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/gappa/gappa_taxonomy.tsv", checkIfExists: true)
                        ]
                    ]
                )
                """

                //FASTA_NEWICK_EPANG_GAPPA ( ch_input )
            }
        }

        then {
            assertAll(
                { assert workflow.success}
            )
        }
    }
}

/**
workflow test_fasta_newick_epang_gappa_mafft {
    
    ch_input = Channel.of(
        [
            meta: [ id: "mafft" ],
            data: [
                alignmethod:  'mafft',
                queryseqfile: file("https://github.com/nf-core/test-datasets/raw/phyloplace/testdata/PF14720_3_sequences.faa", checkIfExists: true),
                refseqfile:   file("https://github.com/nf-core/test-datasets/raw/phyloplace/testdata/PF14720_seed.alnfaa", checkIfExists: true),
                refphylogeny: file("https://github.com/nf-core/test-datasets/raw/phyloplace/testdata/PF14720_seed.ft.LGCAT.newick", checkIfExists: true),
                model:        "LG",
                taxonomy:     file("https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/gappa/gappa_taxonomy.tsv", checkIfExists: true)
            ]
        ]
    )

    FASTA_NEWICK_EPANG_GAPPA ( ch_input )
}

workflow test_fasta_newick_epang_gappa_nucl_hmmer {
    
    ch_input = Channel.of(
        [
            meta: [ id: "nucl_hmmer" ],
            data: [
                alignmethod:  'hmmer',
                queryseqfile: file("https://github.com/nf-core/test-datasets/raw/phyloplace/testdata/cyn_syn.fna", checkIfExists: true),
                refseqfile:   file("https://github.com/nf-core/test-datasets/raw/phyloplace/testdata/cyanos_16s.alnfna", checkIfExists: true),
                refphylogeny: file("https://github.com/nf-core/test-datasets/raw/phyloplace/testdata/cyanos_16s.newick", checkIfExists: true),
                model:        "GTR+F+I+I+R3",
                taxonomy:     []
            ]
        ]
    )

    FASTA_NEWICK_EPANG_GAPPA ( ch_input )
}
**/
