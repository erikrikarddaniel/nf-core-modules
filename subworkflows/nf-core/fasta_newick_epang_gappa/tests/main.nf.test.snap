{
    "hmmer.taxonomy.per_query.tsv": {
        "content": [
            "hmmer.taxonomy.per_query.tsv"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-05T20:45:46.123551197"
    },
    "mafft.taxonomy.profile.tsv_lines": {
        "content": "eb4ae24c330c4b4ab768f7d1c26c7215",
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-08T11:24:06.138041"
    },
    "hmmer.taxonomy.profile.tsv": {
        "content": [
            "hmmer.taxonomy.profile.tsv"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-05T20:45:46.115530176"
    },
    "hmmer.graft.placement.epa_result.newick_lines": {
        "content": "8533b159d543f75b92efcfb4fd330280",
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-08T10:55:06.361779"
    },
    "hmmer.taxonomy.profile.tsv_lines": {
        "content": "8533b159d543f75b92efcfb4fd330280",
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-08T10:59:31.223864"
    },
    "mafft.graft.placement.epa_result.newick_lines": {
        "content": "eb4ae24c330c4b4ab768f7d1c26c7215",
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-08T11:24:06.117996"
    },
    "mafft.taxonomy.per_query.tsv_lines": {
        "content": "eb4ae24c330c4b4ab768f7d1c26c7215",
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-08T11:24:06.175662"
    },
    "hmmer.graft.placement.epa_result.newick": {
        "content": [
            "hmmer.graft.hmmer.epa_result.newick"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-05T20:45:46.105219765"
    },
    "nucl_hmmer.graft.placement.epa_result.newick": {
        "content": [
            "nucl_hmmer.graft.nucl_hmmer.epa_result.newick"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-08T11:37:46.19414"
    },
    "hmmer.epa_result.jplace.gz": {
        "content": [
            "hmmer.epa_result.jplace.gz"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-08T11:56:49.564967"
    },
    "hmmer.taxonomy.per_query.tsv_lines": {
        "content": "8533b159d543f75b92efcfb4fd330280",
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-08T10:59:31.236689"
    },
    "mafft.taxonomy.per_query.tsv": {
        "content": [
            "mafft.taxonomy.per_query.tsv"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-05T20:57:13.505168194"
    },
    "mafft.graft.placement.epa_result.newick": {
        "content": [
            "mafft.graft.mafft.epa_result.newick"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-05T20:57:13.49856233"
    },
    "nucl_hmmer.epa_result.jplace.gz": {
        "content": [
            "nucl_hmmer.epa_result.jplace.gz"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-08T11:44:49.494246"
    },
    "mafft.taxonomy.profile.tsv": {
        "content": [
            "mafft.taxonomy.profile.tsv"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-05T20:57:13.501534227"
    },
    "mafft.epa_result.jplace.gz": {
        "content": [
            "mafft.epa_result.jplace.gz"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-08T11:57:06.815658"
    }
}